io/disk/fiotest.py

```diff
@@ -138,33 +150,27 @@ def setUp(self):
                     if line.startswith(eng) and 'no' in line:
                         self.cancel("PMEM engines not built with fio")
 
-        if not self.disk:
-            self.dir = self.workdir
-
         self.target = self.disk
         self.lv_disk = self.disk
         self.part_obj = Partition(self.disk, mountpoint=self.dir)
         self.sraid = softwareraid.SoftwareRaid(self.raid_name, '0',
                                                self.disk.split(), '1.2')
         dmesg.clear_dmesg()
-        if self.disk in disk.get_disks():
-            self.pre_cleanup()
-            if raid_needed:
-                self.create_raid(self.target, self.raid_name)
-                self.raid_create = True
-                self.target = self.raid_name
-
-            if lv_needed:
-                self.lv_disk = self.target
-                self.target = self.create_lv(self.target)
-                self.lv_create = True
-
-            if fstype:
-                self.create_fs(self.target, self.dir, fstype,
-                               fs_args, mnt_args)
-                self.fs_create = True
-        else:
-            self.cancel("Missing disk %s in OS" % self.disk)
```

Comment:
@AYUSHJAIN951 fiotest is also used for a pmem testing where they have the different ways of using it.
And I don't think the existing way of handling has any problem.

what I feel is you want to extend these test cases to use on OS dir?
if that is the case, please come up with a proper way of handling it instead of just moving the thing up and down.
let me know if you need any help here.

Reply:
```
JOB ID     : 0bfe15e4ed99a1f27d43237a525f79b18fbc9b78
JOB LOG    : /root/tests/results/job-2023-03-07T05.17-0bfe15e/job.log
 (1/1) fiotest.py:FioTest.test;run-dir-disk-fs-ext4-lv-lv-raid-raid-5880: STARTED
 (1/1) fiotest.py:FioTest.test;run-dir-disk-fs-ext4-lv-lv-raid-raid-5880: ERROR: 'NoneType' object has no attribute 'split' (1.12 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/tests/results/job-2023-03-07T05.17-0bfe15e/results.html
JOB TIME   : 2.61 s
```
In the current version of fiotest.py if disk is not provided it fails with this error on line 147 
        self.sraid = softwareraid.SoftwareRaid(self.raid_name, '0',
                                               self.disk.split(), '1.2')
which is needed to be handled thats why pushing disk checks above that

Reply:
I could see it printing the Missing disk message on not providing the disk in yaml.

Reply:
change of this part of the code seems promising, @AYUSHJAIN951 please take care pmem.yaml may we need have align that @Naresh-ibm can you also please help here